prow/pjutil/jobspec.go

```diff
@@ -44,23 +50,60 @@ type JobSpec struct {
 	agent kube.ProwJobAgent
 }
 
-func NewJobSpec(spec kube.ProwJobSpec, buildId string) JobSpec {
+func NewJobSpec(spec kube.ProwJobSpec, buildId, prowJobId string) JobSpec {
 	return JobSpec{
-		Type:    spec.Type,
-		Job:     spec.Job,
-		BuildId: buildId,
-		Refs:    spec.Refs,
-		agent:   spec.Agent,
+		Type:      spec.Type,
+		Job:       spec.Job,
+		BuildId:   buildId,
+		ProwJobId: prowJobId,
+		Refs:      spec.Refs,
+		agent:     spec.Agent,
 	}
 }
 
+// GetBuildID calls out to `tot` in order
+// to vend build identifier for the job
+func GetBuildID(name, totURL string) (string, error) {
+	var err error
+	url, err := url.Parse(totURL)
+	if err != nil {
+		return "", fmt.Errorf("invalid tot url: %v", err)
+	}
+	url.Path = path.Join(url.Path, "vend", name)
+	sleep := 100 * time.Millisecond
+	for retries := 0; retries < 10; retries++ {
```

Comment:
We should start using apimachinery's [ExponentialBackoff](https://github.com/kubernetes/test-infra/blob/6c18b0931506405af718ad3d2ba16e9373ac347d/vendor/k8s.io/apimachinery/pkg/util/wait/wait.go#L192).